### PR TITLE
feat: [CDS-70207]: Update manifests steps strategy type and integrate fetch helm chart manifest changes

### PR DESCRIPTION
**Pull request by abosii-harness**
## Describe your changes
Move ManifestsStepV2 to async strategy and integrate helm chart logic

### Changes as part of this PR:
- Move ManifestsStepV2 to async strategy. The logic should guarantee if the fetch helm chart flow is not enabled then the behavior will be similar with sync. Keeping the old method to ensure backward compatibility in case if deployment was executed with old plan where strategy is still sync
- Integrate the logic to detect if need to fetch manifest metadata and track the task ids

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49989)

### CODE REVIEW

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks, uts

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks, uts

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks, uts

File: 125-cd-nextgen/src/test/java/io/harness/cdng/CDNGTestRule.java [hunk 1]
```diff
@@ -307,6 +307,7 @@ protected void configure() {
         bind(OrganizationService.class).toInstance(mock(OrganizationService.class));
         bind(ProjectService.class).toInstance(mock(ProjectService.class));
         bind(FeatureFlagService.class).toInstance(mock(FeatureFlagServiceImpl.class));
+        bind(CDStepHelper.class).toProvider(() -> mock(CDStepHelper.class)).asEagerSingleton();
```
Comment (abhinavsingh1196) [inline 125-cd-nextgen/src/test/java/io/harness/cdng/CDNGTestRule.java:310]: This change is present in both PRs :)
Is this like a global mock? I see this class is still mocked in ManifestsStepV2Test, that's the only usage I could see in this change.

**Review by abhinavsingh1196 — APPROVED** [review]

File: 125-cd-nextgen/src/test/java/io/harness/cdng/CDNGTestRule.java (hunk 1 above)
Comment (abosii-harness) [reply 125-cd-nextgen/src/test/java/io/harness/cdng/CDNGTestRule.java]: It shouldn't be present in this PR, not sure why you see it. Without this it is causing some dependency injection issues on some of the tests. Can't say what is the overall reason behind this and which changes actually causing this but this is the only solution that I could come up with to fix those errors

**Review by GoelPratyush99 — APPROVED** [review]

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks, uts

**Review by abhinavsingh1196 — DISMISSED** [review]

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks, uts

**Review by abhinavsingh1196 — APPROVED** [review]

**Review by alebuh — APPROVED** [review]

**Review by namantalaycha10 — APPROVED** [review]

**Review by yogesh-chauhan — APPROVED** [review]

**Comment by abosii-harness** [comment]
trigger review

**Pull request merged**